napari/_qt/qt_main_window.py

```diff
@@ -68,40 +67,12 @@ class Window:
     raw_stylesheet = get_stylesheet()
 
     def __init__(self, viewer, *, show: bool = True):
+        # create QApplication if it doesn't already exist
+        app = get_app()
 
-        # Check there is a running app
-        # instance() returns the singleton instance if it exists, or None
-        app = QApplication.instance()
-        # if None, raise a RuntimeError with the appropriate message
-        if app is None:
-            message = (
-                "napari requires a Qt event loop to run. To create one, "
-                "try one of the following: \n"
-                "  - use the `napari.gui_qt()` context manager. See "
-                "https://github.com/napari/napari/tree/master/examples for"
-                " usage examples.\n"
-                "  - In IPython or a local Jupyter instance, use the "
-                "`%gui qt` magic command.\n"
-                "  - Launch IPython with the option `--gui=qt`.\n"
-                "  - (recommended) in your IPython configuration file, add"
-                " or uncomment the line `c.TerminalIPythonApp.gui = 'qt'`."
-                " Then, restart IPython."
-            )
-            raise RuntimeError(message)
-
-        if perf_config:
-            if perf_config.trace_qt_events:
-                from .tracing.qt_event_tracing import convert_app_for_tracing
-
-                # For tracing Qt events we need a special QApplication. If
-                # using `gui_qt` we already have the special one, and no
-                # conversion is done here. However when running inside
-                # IPython or Jupyter this is where we switch out the
-                # QApplication.
-                app = convert_app_for_tracing(app)
-
-            # Will patch based on config file.
-            perf_config.patch_callables()
+        # TODO: We should move this stuff to `get_app` logic
+        # but it's currently tied to attributes on the viewer instance
+        # >>>>>>>>>>>>>>>>>>
```

Comment:
We should ask @Czaki about this, I think he requested adding some of this stuff so he could overwrite them for his dual viewer stuff, but agreed we should find a way to pull off viewer. We can do this in a follow-up PR though if needed

Reply:
Yeah, I think we need to change the API here slightly.  later...

Reply:
This behaviour should be controled from Viewer constructor. As now. Even if napari development allow to create two view as plugin then there will be still need for specialization (to provide some of end user specialized version of application (focused on given task) which will not interact with other instances (not grupping on task bar, different save state location etc). 
To many options may create illegible GUI.

Reply:
Ok, let's discuss this more in a new PR as this one isn't a breaking change here.